<?xml version="1.0"?>
<launch>
  <arg name="point_cloud_in" default="/registration/pc_aligned"/>
  <arg name="enable_action_mode" default="False"/>
  <arg name="debug_mode" default="false"/>
  <arg name="start_manager" default="true"/>
  <arg name="nodelet_manager" default="segmentation_nodelet_manager"/>
  <group if="$(arg start_manager)">
    <node if="$(arg debug_mode)" pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)"  args="manager" output="screen"
          launch-prefix="gdb -ex run --args"/>
    <node unless="$(arg debug_mode)" pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)"  args="manager" output="screen"/>
  </group>
  <node pkg="nodelet" type="nodelet" name="segmentation"
        args="load cob_3d_segmentation/SegmentationAllInOneNodelet $(arg nodelet_manager)" output="screen">
    <!--remap from="cloud_in" to="/cam3d/depth_registered/points"/-->
    <remap from="cloud_in" to="$(arg point_cloud_in)"/>
    <param name="enable_action_mode" type="bool" value="$(arg enable_action_mode)"/>

    <param name="centroid_passthrough" type="double" value="5.0"/>
  </node>
</launch>
